server/common/fbs/matrix.py

```diff
@@ -52,21 +54,22 @@ def serialize_typed_array(builder, source_array, encoding_info):
 
     arr = source_array
     (array_type, as_type) = encoding_info(source_array)
-
     if isinstance(arr, pd.Index):
         arr = arr.to_series()
 
     # convert to a simple ndarray
-    if as_type == "json":
```

Comment:
I need to add this back to catch edge cases (not Categorical and not a typed array, this is probably for columns like cell barcode IDs)